apt-hook/20apt-esm-hook.conf

```diff
@@ -3,5 +3,5 @@ APT::Update::Pre-Invoke {
 };
 
 binary::apt::AptCli::Hooks::Upgrade {
-	"[ ! -f /usr/lib/ubuntu-advantage/apt-esm-json-hook ] || /usr/lib/ubuntu-advantage/apt-esm-json-hook 2> /var/lib/ubuntu-advantage/apt-upgrade-hook-err.txt || true";
+	"[ ! -f /usr/lib/ubuntu-advantage/apt-esm-json-hook ] || /usr/lib/ubuntu-advantage/apt-esm-json-hook 2>> /var/log/ubuntu-advantage-apt-hook.log || true";
```

Comment:
I checked and `debian/ubuntu-pro-client.logrotate` includes this new log file in its existing globbing rule.

Reply:
Did some testing, forcing a logrotate while the log file was empty, and later populating it with some garbage. It worked as expected. Here is when I forcibly populated the file, and then issue logrotate a few times:
```
$ ls -la /var/log/ubuntu-advantage*
-rw-r--r-- 1 root root    0 Sep  4 18:43 /var/log/ubuntu-advantage-apt-hook.log
-rw-r--r-- 1 root root    6 Sep  4 18:43 /var/log/ubuntu-advantage-apt-hook.log.1
-rw-r--r-- 1 root root   26 Sep  4 18:43 /var/log/ubuntu-advantage-apt-hook.log.2.gz
-rw-r--r-- 1 root root    0 Sep  4 18:42 /var/log/ubuntu-advantage.log
-rw-r--r-- 1 root root 2730 Sep  4 18:42 /var/log/ubuntu-advantage.log.1
-rw-r----- 1 root root 7737 Sep  4 18:38 /var/log/ubuntu-advantage.log.2.gz
```
And then the files are zero-sized, as it is in the snapshot above, logrotate correctly does nothing to them.